### PR TITLE
Selecting GPU slice in profile view selects the corresponding command

### PR DESCRIPTION
**Pull request by hliatis**
Bug: b/159482773

### CODE REVIEW

**Comment by tobine** [comment]
I tested this out and it's working correctly for me.
Thanks @hliatis !

File: gapic/src/main/com/google/gapid/perfetto/models/SliceTrack.java [hunk 1]
```diff
@@ -232,7 +232,7 @@ public RenderStageInfo(long frameBufferHandle, String frameBufferName, long rend
   }
 
   public static class Slices implements Selection<Slices> {
-    private int count = 0;
+    protected int count = 0;
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/perfetto/models/SliceTrack.java:235]: Is this necessary? I don't see you using this field?

File: gapic/src/main/com/google/gapid/views/ProfileView.java [hunk 2]
```diff
@@ -343,12 +372,28 @@ protected GpuSliceTrack(long trackId, List<Service.ProfilingData.GpuSlices.Slice
...
+        return new GpuSlices(serverSlices, "GPU Queue Events");
       }
 
-      private static Slices toSlices(List<Service.ProfilingData.GpuSlices.Slice> serverSlices) {
-        return new Slices(serverSlices, "GPU Queue Events");
+      private static class GpuSlices extends Slices {
+        public final List<Long> groupIds = Lists.newArrayList();
+
+        public GpuSlices(List<Service.ProfilingData.GpuSlices.Slice> serverSlices, String title) {
+          super(serverSlices, title);
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/ProfileView.java:387]: Please remove the constructor that takes the proto slices from the superclass and move its logic here.

File: gapic/src/main/com/google/gapid/views/ProfileView.java [hunk 3]
```diff
@@ -343,12 +372,28 @@ protected GpuSliceTrack(long trackId, List<Service.ProfilingData.GpuSlices.Slice
...
+
+          for (Service.ProfilingData.GpuSlices.Slice s : serverSlices) {
+            this.add(s.getGroupId());
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/ProfileView.java:390]: just `groupIds.add(s.getGroupId)` (most likely in the loop you will add by moving the stuff from the superclass constructor.

File: gapic/src/main/com/google/gapid/views/ProfileView.java [hunk 4]
```diff
@@ -137,6 +143,29 @@ public void onProfileLoaded(Loadable.Message error) {
...
+        if (((TraceUi.GpuSliceTrack.GpuSlices)selected).groupIds.get(i) >= 0) {
+          firstGroupId = ((TraceUi.GpuSliceTrack.GpuSlices)selected).groupIds.get(i);
+          break;
+        }
+      }
+
+      if (firstGroupId != -1) {
+        for (Service.ProfilingData.GpuSlices.Group group : models.profile.getData().getSlices().getGroupsList()) {
+          if (firstGroupId == group.getId()) {
+            models.commands.selectCommands(CommandIndex.forCommand(group.getLink()), false);
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/ProfileView.java:162]: also add a `break` here, so we definitely call `selectCommands` only once.

**Pull request merged**